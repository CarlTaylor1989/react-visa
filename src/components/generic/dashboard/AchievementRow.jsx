import React from 'react';
import PropTypes from 'prop-types';
import NumberFormat from 'react-number-format';
import { Translate } from 'react-redux-i18n';
import CompletionBar from '../completionbar/CompletionBar';
import Tooltip from '../tooltip/Tooltip';

import './AchievementRow.scss';

const AchievementRow = (props) => {
  const {
    achievementId,
    complete,
    completion,
    hidden,
    score
  } = props;

  return (
    <div className="achievementRow">
      <div className={`achievementRowIdentifier ${complete ? 'complete' : ''}`}>
        <div className={`achievementRowIcon ${achievementId} ${complete ? 'dark' : ''}`} />
        <div className="achievementRowInformation">
          <Tooltip fieldName="generic.tooltips.dashboard.achievements.title">
            <div className="achievementRowName">
              <Translate value={`generic.achievements.${achievementId}.title`} />
            </div>
          </Tooltip>
          {!complete && hidden ? (
            <Tooltip fieldName="generic.tooltips.dashboard.achievements.hidden">
              <div className="achievementRowDescription">
                <Translate value="generic.dashboard.hidden" />
              </div>
            </Tooltip>
          ) : (
            <Tooltip fieldName="generic.tooltips.dashboard.achievements.description">
              <div className="achievementRowDescription">
                <Translate value={`generic.achievements.${achievementId}.description`} />
              </div>
            </Tooltip>
          )}
        </div>
      </div>
      <div className="achievementRowProgress">
        <Tooltip fieldName="generic.tooltips.dashboard.achievements.progress">
          <div className="achievementRowCompletion">
            <CompletionBar
              {...completion}
              displayValue={!(!complete && hidden)}
              hiddenTextField="generic.dashboard.hiddenValue"
            />
          </div>
        </Tooltip>
        <Tooltip fieldName="generic.tooltips.dashboard.achievements.points">
          <div className={`achievementRowScore ${complete ? 'complete' : ''}`}>
            <Translate value="generic.dashboard.achievementPlus" />
            <NumberFormat
              value={score}
              displayType="text"
              thousandSeparator=","
            />
          </div>
        </Tooltip>
      </div>
    </div>
  );
};

AchievementRow.propTypes = {
  achievementId: PropTypes.string.isRequired,
  complete: PropTypes.bool.isRequired,
  completion: PropTypes.object.isRequired,
  hidden: PropTypes.bool.isRequired,
  score: PropTypes.number.isRequired
};

export default AchievementRow;
