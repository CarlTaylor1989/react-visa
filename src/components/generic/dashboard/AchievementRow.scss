@import '../../../styles/variables';
@import '../../../styles/mixins';

.achievementRow {
  display: flex;
  flex-direction: column;

  .achievementRowIdentifier {
    @include align-items(flex-start);
    display: flex;

    &.complete {
      .achievementRowName {
        color: $colour-yellow;
      }
    }

    .achievementRowInformation {
      display: flex;
      flex-direction: column;
      padding: 7px 0 0 8px;

      .achievementRowName {
        font-size: 24px;
        font-weight: 800;
        line-height: 24px;
      }

      .achievementRowDescription {
        font-size: 20px;
      }
    }
  }

  .achievementRowProgress {
    @include align-items(center);
    display: flex;

    .achievementRowCompletion {
      margin-left: 44px;

      .completionBar {
        width: 1006px;
      }
    }

    .achievementRowScore {
      @include align-items(center);
      @include border-radius(24px);
      @include multi-angle-gradient(
        152deg,
        rgba($colour-white, 0) 0%,
        rgba($colour-white, .01) 9%,
        rgba($colour-white, .05) 23%,
        rgba($colour-white, .1) 43%,
        rgba($colour-white, .18) 66%,
        rgba($colour-white, .27) 92%,
        rgba($colour-white, .3) 100%
      );
      border: 2px solid $colour-white;
      color: $colour-yellow;
      font-size: 32px;
      font-weight: 700;
      height: 48px;
      margin-left: 45px;
      opacity: .3;
      text-align: center;
      width: 223px;

      &.complete {
        opacity: 1;
      }
    }
  }
}
