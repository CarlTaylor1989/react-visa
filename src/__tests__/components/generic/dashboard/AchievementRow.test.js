/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import AchievementRow from '../../../../components/generic/dashboard/AchievementRow';

let mockProps;
beforeEach(() => {
  mockProps = {
    achievementId: 'ac1',
    complete: false,
    completion: {
      percent: 50,
      amount: 5000,
      total: 10000
    },
    hidden: false,
    score: 10000
  };
});

test('renders the AchievementRow component with an incomplete achievement', () => {
  const wrapper = shallow(<AchievementRow {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders the AchievementRow component with a incomplete achievement', () => {
  const wrapper = shallow(<AchievementRow {...mockProps} complete />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders the AchievementRow component with a hidden achievement', () => {
  const wrapper = shallow(<AchievementRow {...mockProps} hidden />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});
