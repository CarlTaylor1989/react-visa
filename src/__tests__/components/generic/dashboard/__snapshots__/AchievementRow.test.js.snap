// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the AchievementRow component with a hidden achievement 1`] = `
<div
  className="achievementRow"
>
  <div
    className="achievementRowIdentifier "
  >
    <div
      className="achievementRowIcon ac1 "
    />
    <div
      className="achievementRowInformation"
    >
      <Connect(Tooltip)
        fieldName="generic.tooltips.dashboard.achievements.title"
      >
        <div
          className="achievementRowName"
        >
          <Translate
            tag="span"
            value="generic.achievements.ac1.title"
          />
        </div>
      </Connect(Tooltip)>
      <Connect(Tooltip)
        fieldName="generic.tooltips.dashboard.achievements.hidden"
      >
        <div
          className="achievementRowDescription"
        >
          <Translate
            tag="span"
            value="generic.dashboard.hidden"
          />
        </div>
      </Connect(Tooltip)>
    </div>
  </div>
  <div
    className="achievementRowProgress"
  >
    <Connect(Tooltip)
      fieldName="generic.tooltips.dashboard.achievements.progress"
    >
      <div
        className="achievementRowCompletion"
      >
        <CompletionBar
          amount={5000}
          displayValue={false}
          hiddenTextField="generic.dashboard.hiddenValue"
          percent={50}
          total={10000}
        />
      </div>
    </Connect(Tooltip)>
    <Connect(Tooltip)
      fieldName="generic.tooltips.dashboard.achievements.points"
    >
      <div
        className="achievementRowScore "
      >
        <Translate
          tag="span"
          value="generic.dashboard.achievementPlus"
        />
        <NumberFormat
          allowEmptyFormatting={false}
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          fixedDecimalScale={false}
          getInputRef={[Function]}
          isAllowed={[Function]}
          isNumericString={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          onValueChange={[Function]}
          prefix=""
          suffix=""
          thousandSeparator=","
          thousandsGroupStyle="thousand"
          type="text"
          value={10000}
        />
      </div>
    </Connect(Tooltip)>
  </div>
</div>
`;

exports[`renders the AchievementRow component with a incomplete achievement 1`] = `
<div
  className="achievementRow"
>
  <div
    className="achievementRowIdentifier complete"
  >
    <div
      className="achievementRowIcon ac1 dark"
    />
    <div
      className="achievementRowInformation"
    >
      <Connect(Tooltip)
        fieldName="generic.tooltips.dashboard.achievements.title"
      >
        <div
          className="achievementRowName"
        >
          <Translate
            tag="span"
            value="generic.achievements.ac1.title"
          />
        </div>
      </Connect(Tooltip)>
      <Connect(Tooltip)
        fieldName="generic.tooltips.dashboard.achievements.description"
      >
        <div
          className="achievementRowDescription"
        >
          <Translate
            tag="span"
            value="generic.achievements.ac1.description"
          />
        </div>
      </Connect(Tooltip)>
    </div>
  </div>
  <div
    className="achievementRowProgress"
  >
    <Connect(Tooltip)
      fieldName="generic.tooltips.dashboard.achievements.progress"
    >
      <div
        className="achievementRowCompletion"
      >
        <CompletionBar
          amount={5000}
          displayValue={true}
          hiddenTextField="generic.dashboard.hiddenValue"
          percent={50}
          total={10000}
        />
      </div>
    </Connect(Tooltip)>
    <Connect(Tooltip)
      fieldName="generic.tooltips.dashboard.achievements.points"
    >
      <div
        className="achievementRowScore complete"
      >
        <Translate
          tag="span"
          value="generic.dashboard.achievementPlus"
        />
        <NumberFormat
          allowEmptyFormatting={false}
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          fixedDecimalScale={false}
          getInputRef={[Function]}
          isAllowed={[Function]}
          isNumericString={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          onValueChange={[Function]}
          prefix=""
          suffix=""
          thousandSeparator=","
          thousandsGroupStyle="thousand"
          type="text"
          value={10000}
        />
      </div>
    </Connect(Tooltip)>
  </div>
</div>
`;

exports[`renders the AchievementRow component with an incomplete achievement 1`] = `
<div
  className="achievementRow"
>
  <div
    className="achievementRowIdentifier "
  >
    <div
      className="achievementRowIcon ac1 "
    />
    <div
      className="achievementRowInformation"
    >
      <Connect(Tooltip)
        fieldName="generic.tooltips.dashboard.achievements.title"
      >
        <div
          className="achievementRowName"
        >
          <Translate
            tag="span"
            value="generic.achievements.ac1.title"
          />
        </div>
      </Connect(Tooltip)>
      <Connect(Tooltip)
        fieldName="generic.tooltips.dashboard.achievements.description"
      >
        <div
          className="achievementRowDescription"
        >
          <Translate
            tag="span"
            value="generic.achievements.ac1.description"
          />
        </div>
      </Connect(Tooltip)>
    </div>
  </div>
  <div
    className="achievementRowProgress"
  >
    <Connect(Tooltip)
      fieldName="generic.tooltips.dashboard.achievements.progress"
    >
      <div
        className="achievementRowCompletion"
      >
        <CompletionBar
          amount={5000}
          displayValue={true}
          hiddenTextField="generic.dashboard.hiddenValue"
          percent={50}
          total={10000}
        />
      </div>
    </Connect(Tooltip)>
    <Connect(Tooltip)
      fieldName="generic.tooltips.dashboard.achievements.points"
    >
      <div
        className="achievementRowScore "
      >
        <Translate
          tag="span"
          value="generic.dashboard.achievementPlus"
        />
        <NumberFormat
          allowEmptyFormatting={false}
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          fixedDecimalScale={false}
          getInputRef={[Function]}
          isAllowed={[Function]}
          isNumericString={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          onValueChange={[Function]}
          prefix=""
          suffix=""
          thousandSeparator=","
          thousandsGroupStyle="thousand"
          type="text"
          value={10000}
        />
      </div>
    </Connect(Tooltip)>
  </div>
</div>
`;
